tests/validator/test_validation_graph.py

```diff
@@ -178,10 +249,226 @@ def test_ExpectationValidationGraph_get_exception_info(
+
+
+@pytest.mark.unit
+def test_parse_validation_graph(
+    expect_column_value_z_scores_to_be_less_than_expectation_validation_graph: ValidationGraph,
+):
+    available_metric: Dict[Tuple[str, str, str], Any]
+
+    # Parse input "ValidationGraph" object and confirm the numbers of ready and still needed metrics.
+    available_metrics = {}
+    (
+        ready_metrics,
+        needed_metrics,
+    ) = expect_column_value_z_scores_to_be_less_than_expectation_validation_graph._parse(
+        metrics=available_metrics
+    )
+    assert len(ready_metrics) == 2 and len(needed_metrics) == 9
+
+    # Show that including "nonexistent" metric in dictionary of resolved metrics does not increase ready_metrics count.
+    available_metrics = {("nonexistent", "nonexistent", "nonexistent"): "NONE"}
+    (
+        ready_metrics,
+        needed_metrics,
+    ) = expect_column_value_z_scores_to_be_less_than_expectation_validation_graph._parse(
+        metrics=available_metrics
+    )
+    assert len(ready_metrics) == 2 and len(needed_metrics) == 9
+
+
+@pytest.mark.unit
+def test_populate_dependencies(
+    expect_column_value_z_scores_to_be_less_than_expectation_validation_graph: ValidationGraph,
+):
+    assert (
+        len(
+            expect_column_value_z_scores_to_be_less_than_expectation_validation_graph.edges
+        )
+        == 33
+    )
+
+
+@pytest.mark.unit
+def test_populate_dependencies_with_incorrect_metric_name():
+    class PandasExecutionEngineStub:
+        pass
+
+    PandasExecutionEngineStub.__name__ = "PandasExecutionEngine"
+    pandas_execution_engine_stub = cast(ExecutionEngine, PandasExecutionEngineStub())
+
+    graph = ValidationGraph(execution_engine=pandas_execution_engine_stub)
+    try:
+        graph.build_metric_dependency_graph(
+            metric_configuration=MetricConfiguration(
+                "column_values.not_a_metric", IDDict()
+            ),
+        )
+    except ge_exceptions.MetricProviderError as e:
+        graph = e
+
+    assert isinstance(graph, ge_exceptions.MetricProviderError)
+
+
+@pytest.mark.integration
+def test_resolve_validation_graph_with_bad_config_catch_exceptions_true(
+    basic_datasource,
+):
+    df = pd.DataFrame({"a": [1, 5, 22, 3, 5, 10], "b": [1, 2, 3, 4, 5, None]})
+
+    batch = basic_datasource.get_single_batch_from_batch_request(
+        RuntimeBatchRequest(
+            **{
+                "datasource_name": "my_datasource",
+                "data_connector_name": "test_runtime_data_connector",
+                "data_asset_name": "IN_MEMORY_DATA_ASSET",
+                "runtime_parameters": {
+                    "batch_data": df,
+                },
+                "batch_identifiers": {
+                    "pipeline_stage_name": 0,
+                    "airflow_run_id": 0,
+                    "custom_key_0": 0,
+                },
+            }
+        )
+    )
+
+    expectation_configuration = ExpectationConfiguration(
```

Comment:
is this meant to be the aborted metric? consider naming `bad_configuration`

Reply:
@donaldheppner This is meant to indicate an `ExpectationConfiguration`, which will depend on a metric that will fail, because it will be asked to access a column that does not exist.  Do we still want to change anything in the variable names?  I feel it is not necessary, but happy to discuss.  Thanks.